Lib/unittest/test/testmock/testasync.py

```diff
@@ -50,6 +52,17 @@ def a(self):
 normal_foo_name = f'{__name__}.NormalClass'
 
 
+@contextmanager
+def assertNeverAwaited(test):
+    with test.assertWarns(RuntimeWarning) as cm:
+        yield
+        # In non-CPython implementations of Python, this is needed because timely
+        # deallocation is not guaranteed by the garbage collector.
```

Comment:
Do we need to add a check to gc only if it's not CPython implementation or narrow it only to PyPy ?

Reply:
Nope, there's zero downside to calling gc.collect() here. It's way cpython currently does as an implementation detail, but these tests are relying on it...